### DISCUSSION TITLE
IBS P01B Inkbird swimming pool sensor

### DISCUSSION THREAD

**Discussion by 1technophile** [theengs/decoder#131] [New devices/things integration]
Opening this thread following this request:
https://community.home-assistant.io/t/passive-ble-monitor-integration/303583/475?u=1technophile

What we need to be able to integrate the sensor:
3 screenshots from nrf connect mobile app like below:
![Screenshot_20220619-120324_nRF Connect~2](https://user-images.githubusercontent.com/12672732/175518188-aa11e3a5-cd87-4b88-b546-65d80c3e01b8.jpg)

The 3 corresponding pictures (same time of the screenshots) of the values seen on the devices.

The values needs to be different from one picture to another.

**Comment by Azelphur** [comment]
Hi,

My original issue is solved with the Passive BLE Tracker but I'm happy to contribute the information you need to get this going here! :)

I installed the nrf connect app and found the IBS P01B, however, on that page the number (0xFFFF009FD30008) on the scanner tab never seem to change.

![Screenshot_20220624-115432_nRF Connect](https://user-images.githubusercontent.com/280694/175522265-c53cd88d-b7a1-4347-96b9-31517094de68.png)

I tried clicking connect and I got this

![Screenshot_20220624-115402_nRF Connect](https://user-images.githubusercontent.com/280694/175522326-7aff7756-edea-4bc9-b435-9a0276fce28a.png)

Let me know if there's anything else I can do :)

**Reply by 1technophile** [reply]
Thanks for sharing, seems good data. What do you see on the device at the same time?
You don't need to connect to the device to get the manufacturer data, just the screenshot you have is good.

**Comment by Azelphur** [comment]
![IMG_20220624_130720](https://user-images.githubusercontent.com/280694/175531478-3da5f805-c063-44d4-ba8f-d4d71f6c4db2.jpg)

Here's the display of the device, it always shows 0% battery, I think that's because I'm using rechargeable batteries.

**Reply by 1technophile** [reply]
0A76 =2678 = 26.78 ° for the exterior temperature

Could you take another nrf connect screenshot please, not sure of the pool temperature

**Reply by Azelphur** [reply]
That sounds correct to me, when I was using NRF the sensor was on my desk so reading air temp. Back in the pool it's reading 0x0ABB = 2747 = 27.47

![Screenshot_20220624-142518_nRF Connect](https://user-images.githubusercontent.com/280694/175545564-e5842a15-25b5-41e0-9dee-4e26acc16f34.png)

**Reply by 1technophile** [reply]
0x0ABB = 27.47C is the pool temperature now ?
I mean is there 2 probes on this sensor (air and water) or only one ?

I suppose that your battery is at 8% ?

**Reply by Azelphur** [reply]
Sorry slow reply, afaik there's only one sensor, only one shows up in the official app. I have no idea what the second temperature readout on the display is, it's undocumented.

8% maybe, according to the Passive BLE Tracker it's 0%, I think it shows weird values because I'm using rechargeable batteries.

**Reply by DigiH** [reply]
Looks as if the second temperature is the MIN or MAX temperature value, depending for ALL-TIME or 24 HOURS, as stated in the brief manual

https://drive.google.com/file/d/16vlqjqAEw_lkXXx8CIZElgEm2N2aQd2M/view
from
https://inkbird.com/pages/download

Not sure if these values are also broadcast, or possibly even the different ALL-TIME or 24 HOURS setting for them.

**Reply by DigiH** [reply]
The only other two possible MIN/MAX temps I can see in the above two examples are 

9F D3
and
B2 03

while the former could well indicate 9F (15.9C) MIN temp and D3 (21.1ºC) and still doesn't tally with the current 26.7ºC current temperature, and the latter even more confusing, especially with 03 (3ºC), even for Kent ;), I also think I can just about make out that you have it set to ALL-TIME on the photo.

One option would be to reset the MIN MAX temperatures, as described in the manual, as then for a while the current, max and min temperatures should all be the same. Similarly trying to change the min/max display from all-time to last 24 hours, just to see if any current **00** hex byte might change to **01** or such.

**Reply by DigiH** [reply]
More likely though that both bytes show a temperature and since the actual display on the P01B seems to periodically switch from displaying the minimum or the maximum temperature (MIN in the photo I would think) as the second temperature, I would assume the broadcast might follow this as well.

**Comment by Azelphur** [comment]
![IMG_20220628_080707](https://user-images.githubusercontent.com/280694/176116175-80bd8570-eed2-490b-9816-e07f4c9feb2d.jpg)
![Screenshot_20220628-080444_Engbird](https://user-images.githubusercontent.com/280694/176116186-f35f494f-1c05-45a8-bfbb-f862d2b7bf9f.png)
![Screenshot_20220628-080610_nRF Connect](https://user-images.githubusercontent.com/280694/176116189-1c4cf40f-305c-4ad9-b328-0a7acdaacbd9.png)
![Screenshot_20220628-080617_nRF Connect](https://user-images.githubusercontent.com/280694/176116192-e3a478f9-8c17-4acf-88b7-e86adff22cca.png)

Took a whole new set of screenshots, hopefully this will help narrow it down. They were all taken within around a minute of each other. Will grab one after doing a min/max display change soon.

**Reply by 1technophile** [reply]
Hello,

Did you have the opportunity to do your min max display change ?

**Reply by Azelphur** [reply]
Apologies, had a few busy days forgot about this, but here's the information. I pulled the thermometer out of the pool so I could get better pictures.

First, baseline, no changes, set to all time
Max:
![IMG_20220704_134029](https://user-images.githubusercontent.com/280694/177157819-b1891cf6-ba5f-494c-8395-76e1995fea65.jpg)

Min:
![IMG_20220704_134032](https://user-images.githubusercontent.com/280694/177157848-0ac686e2-1507-4665-b006-75cbf2144dd7.jpg)

NRF Connect:
![Screenshot_20220704-134104_nRF Connect](https://user-images.githubusercontent.com/280694/177157926-aee7bb98-00f1-466a-a2c0-0c6bbf3b62fb.png)

Next I changed it to 24 hours mode:

Max:
![IMG_20220704_134139](https://user-images.githubusercontent.com/280694/177158071-1ec3ddb0-454d-42c1-a61c-6b51eccb9121.jpg)

Min:
![IMG_20220704_134150](https://user-images.githubusercontent.com/280694/177158155-de8df0a9-7e5b-4292-9a45-e5c3e06048fa.jpg)

NRF Connect:
![Screenshot_20220704-134159_nRF Connect](https://user-images.githubusercontent.com/280694/177158232-b37f4841-1f53-47cf-8703-1540505c81cd.png)

Then I cleared the data:

Max:
![IMG_20220704_134224](https://user-images.githubusercontent.com/280694/177158401-11661b8d-f4e0-41ea-a868-e262c213299a.jpg)

Min:
![IMG_20220704_134226](https://user-images.githubusercontent.com/280694/177158430-af405347-61db-4656-9b7b-5f6a09670ccd.jpg)

NRF Connect:
![Screenshot_20220704-134238_nRF Connect](https://user-images.githubusercontent.com/280694/177158484-1c4f1e67-6384-4683-bb05-e653eeee7a3e.png)

Then I switched it back to ALL-TIME mode:

Max:
![IMG_20220704_134257](https://user-images.githubusercontent.com/280694/177158518-737c3e8b-20f9-4e5c-90e7-13034d905b49.jpg)

Min:
![IMG_20220704_134259](https://user-images.githubusercontent.com/280694/177158545-946d39b1-c323-4c3c-8518-709c897cc928.jpg)

NRF Connect:
![Screenshot_20220704-134310_nRF Connect](https://user-images.githubusercontent.com/280694/177158573-8f77ed19-590f-4d96-8123-fcc2fa8cd895.png)

**Reply by DigiH** [reply]
Many thanks @Azelphur! While index 10-13 constantly change there is no apparent, at least to me so far, relation to the MIN/MAX temperatures and settings.

The current temperature and battery should already be recognised in Decoder as the IBS-TH2 as they share identical characteristics, including the broadcast `'tps'` name.
